mono/mini/mini-profiler.c

```diff
@@ -150,13 +150,13 @@ mini_profiler_emit_tail_call (MonoCompile *cfg, MonoMethod *target)
 	EMIT_NEW_PCONST (cfg, iargs [1], NULL);
 
 	if (target)
-		EMIT_NEW_METHODCONST (cfg, iargs [2], target);
+		EMIT_NEW_METHODCONST (cfg, iargs [2], target); 
 	else
 		EMIT_NEW_PCONST (cfg, iargs [2], NULL);
 
 	/* void mono_profiler_raise_method_tail_call (MonoMethod *method, MonoMethod *target) */
 	if (trace)
-		mono_emit_jit_icall (cfg, mono_trace_leave_method, iargs);
+		mono_emit_jit_icall (cfg, mono_trace_tail_method, iargs);
```

Comment:
why is this changed ?

Reply:
The explanation is in the text above but briefly. mini-profiler currently sets up the call to mono_trace_leave_method with the two methods as the parameters. However, mono_trace_leave_method expects the 2nd parameter to be a MonoProfilerCallContext from which it extracts the return MonoType. So (a) this parameter is incorrect for tail profiling and (b) it makes no sense to process the return variable when there is no return data for a tail call (there is one when we leave the method that was tail called). 

This new call with process the two method parameters passed and report it using the information in those two method parameters.